fulp_modules/lisa/joel_gun.dm

```diff
@@ -0,0 +1,120 @@
+	..()
+	if(!istype(A, /obj/item/ammo_casing))
+		return
+	user.visible_message("<span class='danger'>[user.name] puts the bullet in [src]!</span>",\
+	 "<span class='userdanger'>You put the bullet in [src]!</span>",\
+	 "<span class='hear'>You hear metal clanking...</span>")
+	var/mob/living/carbon/C = user
+	C.adjustBruteLoss(-10)
+	update_appearance()
+	A.update_appearance()
+	return
+
+/// Gun Reveal + Gun Toss
+/obj/item/gun/ballistic/revolver/joel/attack_self(mob/user)
+	if(!used_ability)
+		if(HAS_TRAIT(user, TRAIT_GUNFLIP)) /// If you have a holster, use gun reveal.
+			user.visible_message("<span class='danger'>[user.name] opens their holster and pulls out [src]!</span>",\
+			 "<span class='userdanger'>You reveal [src] from your holster!</span>",\
+			 "<span class='hear'>You hear metal clanking...</span>")
+			for(var/mob/living/carbon/human/L in viewers(7, user))
+				if(prob(15) && !(L == user) && (L.stat))
+					to_chat(L, "<span class='warning'>Seeing [src] revealed in such a manner disgusts you!</span>")
+					if(!L.mind || !istype(L.mind.martial_art, /datum/martial_art/velvetfu))
+						L.vomit()
+			playsound(src, 'fulp_modules/lisa/Sounds/gunreveal.ogg', 20, FALSE, -5)
+			used_ability = TRUE
+			addtimer(CALLBACK(src, .proc/clear_cooldown), 5 SECONDS)
+			return
+		else /// Otherwise, use Gun Toss.
+			user.visible_message("<span class='danger'>[user.name] throws [src] around!</span>",\
+			 "<span class='userdanger'>You toss [src] around!</span>",\
+			 "<span class='hear'>You hear metal clanking...</span>")
+			for(var/mob/living/L in viewers(7, user))
+				L.face_atom(user)
+				L.do_alert_animation()
+			playsound(src, 'fulp_modules/lisa/Sounds/guntoss.ogg', 50, FALSE, -5)
+			used_ability = TRUE
+			addtimer(CALLBACK(src, .proc/clear_cooldown), 5 SECONDS)
+			var/mob/living/carbon/C = user
+			C.adjustStaminaLoss(-25)
+			C.drop_all_held_items()
+			return
+	..()
+
+/// Steady Aim
+/obj/item/gun/ballistic/revolver/joel/attack_secondary(mob/living/victim, mob/living/user, params)
+	if(used_ability)
+		to_chat(user, "<span class='warning'>You have to wait before using an ability!</span>")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(user == victim)
+		to_chat(user,"<span class='warning'>You can't steady aim yourself!</span>")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	user.visible_message("<span class='danger'>[user.name] holds [victim.name] at gunpoint!</span>",\
+	 "<span class='userdanger'>You hold [victim.name] up at gunpoint!</span>",\
+	 "<span class='hear'>You hear metal clanking...</span>")
+	playsound(loc, 'fulp_modules/lisa/Sounds/steadyaim.ogg', 50, FALSE, -5)
+	if(prob(50))
+		victim.Stun(rand(10, 25))
+	used_ability = TRUE
+	addtimer(CALLBACK(src, .proc/clear_cooldown), 5 SECONDS)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/// Prevents the gun from being fired.
+/obj/item/gun/ballistic/revolver/joel/afterattack(atom/target, mob/living/user, flag, params)
+	. = ..(null, user, flag, params)
```

Comment:
... won't this run the code for firing the the gun?

also, have you accounted for dual-wielding?

Reply:
This part was copy pasted from russian revolver, not sure about dual-wielding.